File: SNI_UI2/wwwroot/Model/DBOViewModel.js
import { EntityClass } from "../WDevCore/WModules/EntityClass.js";
import { WAjaxTools } from "../WDevCore/WModules/WComponentsTools.js";

class ViewParticipantesProyectos extends EntityClass {
   constructor(props) {
       super(props, 'ViewDBO');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   Id_Investigador = { type: 'number' };
   Id_Proyecto = { type: 'number' };
   Fecha_Ingreso = { type: 'date' };
   Estado_Participante = { type: 'text' };
   DescripcionProyecto = { type: 'text' };
   Fecha_Inicio = { type: 'date' };
   Fecha_Finalizacion = { type: 'date' };
   Visibilidad = { type: 'text' };
   Cargo = { type: 'text' };
   Descripcion_Tipo_Proyecto = { type: 'text' };
   Estado_Tipo_Proyecto = { type: 'text' };
   Nombre_Proyecto = { type: 'text' };
   Estado_Proyecto = { type: 'text' };
}
export { ViewParticipantesProyectos }

class ViewCalendarioByDependencia extends EntityClass {
   constructor(props) {
       super(props, 'ViewDBO');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   IdActividad = { type: 'number' };
   IdTareaPadre = { type: 'number' };
   Fecha_Inicial = { type: 'date' };
   Fecha_Final = { type: 'date' };
   Estado = { type: 'text' };
   IdCalendario = { type: 'number' };
   IdTarea = { type: 'number' };
   Id_Dependencia = { type: 'number' };
}
export { ViewCalendarioByDependencia }
class ViewActividadesParticipantes extends EntityClass {
   constructor(props) {
       super(props, 'ViewDBO');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   IdActividad = { type: 'number' };
   Titulo = { type: 'text' };
   Descripcion = { type: 'text' };
   Estado = { type: 'text' };
   Id_Investigador = { type: 'number' };
}
export { ViewActividadesParticipantes }

File: SNI_UI2/CAPA_NEGOCIO/Mapping/Entity/DBOViewModel.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {

   public class ViewParticipantesProyectos : EntityClass {
       public int? Id_Investigador { get; set; }
       public int? Id_Proyecto { get; set; }
       public DateTime? Fecha_Ingreso { get; set; }
       public string? Estado_Participante { get; set; }
       public string? DescripcionProyecto { get; set; }
       public DateTime? Fecha_Inicio { get; set; }
       public DateTime? Fecha_Finalizacion { get; set; }
       public string? Visibilidad { get; set; }
       public string? Cargo { get; set; }
       public string? Descripcion_Tipo_Proyecto { get; set; }
       public string? Estado_Tipo_Proyecto { get; set; }
       public string? Nombre_Proyecto { get; set; }
       public string? Estado_Proyecto { get; set; }
   }
   
   public class ViewCalendarioByDependencia : EntityClass {
       public int? IdActividad { get; set; }
       public int? IdTareaPadre { get; set; }
       public DateTime? Fecha_Inicial { get; set; }
       public DateTime? Fecha_Final { get; set; }
       public string? Estado { get; set; }
       public int? IdCalendario { get; set; }
       public int? IdTarea { get; set; }
       public int? Id_Dependencia { get; set; }
   }
   public class ViewActividadesParticipantes : EntityClass {
       public int? IdActividad { get; set; }
       public string? Titulo { get; set; }
       public string? Descripcion { get; set; }
       public string? Estado { get; set; }
       public int? Id_Investigador { get; set; }
   }
}
